---
pageClass: "rule-details"
sidebarDepth: 0
title: "markdown-preferences/indent"
description: "enforce consistent indentation in Markdown files"
---

# markdown-preferences/indent

> enforce consistent indentation in Markdown files

- â— <badge text="This rule has not been released yet." vertical="middle" type="error"> **_This rule has not been released yet._** </badge>
- âš™ï¸ This rule is included in `plugin.configs.standard`.
- ğŸ”§ The `--fix` option on the [command line](https://eslint.org/docs/user-guide/command-line-interface#fixing-problems) can automatically fix some of the problems reported by this rule.

## ğŸ“– Rule Details

This rule reports ???.

<!-- prettier-ignore-start -->

<!-- eslint-skip -->

```md
<!-- eslint markdown-preferences/indent: 'error' -->

<!-- âœ“ GOOD -->


<!-- âœ— BAD -->

```

<!-- prettier-ignore-end -->

## ğŸ”§ Options

This rule has no options.

<!-- or -->

```json
{
  "markdown-preferences/indent": ["error", {}]
}
```

-

## ğŸ“š Further Reading

-

## ğŸ‘« Related Rules

- [markdown-preferences/blockquote-marker-alignment](./blockquote-marker-alignment.md)
- [markdown-preferences/list-marker-alignment](./list-marker-alignment.md)

## ğŸ” Implementation

<!-- eslint-disable markdown-links/no-dead-urls -- Auto generated -->

- [Rule source](https://github.com/ota-meshi/eslint-plugin-markdown-preferences/blob/main/src/rules/indent.ts)
- [Test source](https://github.com/ota-meshi/eslint-plugin-markdown-preferences/blob/main/tests/src/rules/indent.ts)
- [Test fixture sources](https://github.com/ota-meshi/eslint-plugin-markdown-preferences/tree/main/tests/fixtures/rules/indent)

<!-- eslint-enable markdown-links/no-dead-urls -- Auto generated -->
