---
pageClass: "rule-details"
sidebarDepth: 0
title: "markdown-preferences/indent"
description: "enforce consistent indentation in Markdown files"
---

# markdown-preferences/indent

> enforce consistent indentation in Markdown files

- ❗ <badge text="This rule has not been released yet." vertical="middle" type="error"> **_This rule has not been released yet._** </badge>
- ⚙️ This rule is included in `plugin.configs.standard`.
- 🔧 The `--fix` option on the [command line](https://eslint.org/docs/user-guide/command-line-interface#fixing-problems) can automatically fix some of the problems reported by this rule.

## 📖 Rule Details

This rule reports ???.

<!-- prettier-ignore-start -->

<!-- eslint-skip -->

```md
<!-- eslint markdown-preferences/indent: 'error' -->

<!-- ✓ GOOD -->


<!-- ✗ BAD -->

```

<!-- prettier-ignore-end -->

## 🔧 Options

This rule has no options.

<!-- or -->

```json
{
  "markdown-preferences/indent": ["error", {}]
}
```

-

## 📚 Further Reading

-

## 👫 Related Rules

- [markdown-preferences/blockquote-marker-alignment](./blockquote-marker-alignment.md)
- [markdown-preferences/list-marker-alignment](./list-marker-alignment.md)

## 🔍 Implementation

<!-- eslint-disable markdown-links/no-dead-urls -- Auto generated -->

- [Rule source](https://github.com/ota-meshi/eslint-plugin-markdown-preferences/blob/main/src/rules/indent.ts)
- [Test source](https://github.com/ota-meshi/eslint-plugin-markdown-preferences/blob/main/tests/src/rules/indent.ts)
- [Test fixture sources](https://github.com/ota-meshi/eslint-plugin-markdown-preferences/tree/main/tests/fixtures/rules/indent)

<!-- eslint-enable markdown-links/no-dead-urls -- Auto generated -->
