---
pageClass: "rule-details"
sidebarDepth: 0
title: "markdown-preferences/link-paren-spacing"
description: "enforce consistent spacing inside link parentheses"
---

# markdown-preferences/link-paren-spacing

> enforce consistent spacing inside link parentheses

- â— <badge text="This rule has not been released yet." vertical="middle" type="error"> **_This rule has not been released yet._** </badge>
- âš™ï¸ This rule is included in `plugin.configs.standard`.
- ğŸ”§ The `--fix` option on the [command line](https://eslint.org/docs/user-guide/command-line-interface#fixing-problems) can automatically fix some of the problems reported by this rule.

## ğŸ“– Rule Details

This rule reports ???.

<!-- prettier-ignore-start -->

<!-- eslint-skip -->

```md
<!-- eslint markdown-preferences/link-paren-spacing: 'error' -->

<!-- âœ“ GOOD -->


<!-- âœ— BAD -->

```

<!-- prettier-ignore-end -->

## ğŸ”§ Options

This rule has no options.

<!-- or -->

```json
{
  "markdown-preferences/link-paren-spacing": ["error", {}]
}
```

-

## ğŸ“š Further Reading

-

## ğŸ‘« Related Rules

- [xxx]

[xxx]: https://xxx

## ğŸ” Implementation

<!-- eslint-disable markdown-links/no-dead-urls -- Auto generated -->

- [Rule source](https://github.com/ota-meshi/eslint-plugin-markdown-preferences/blob/main/src/rules/link-paren-spacing.ts)
- [Test source](https://github.com/ota-meshi/eslint-plugin-markdown-preferences/blob/main/tests/src/rules/link-paren-spacing.ts)
- [Test fixture sources](https://github.com/ota-meshi/eslint-plugin-markdown-preferences/tree/main/tests/fixtures/rules/link-paren-spacing)

<!-- eslint-enable markdown-links/no-dead-urls -- Auto generated -->
