#  (markdown-preferences/max-len)

> description

## ðŸ“– Rule Details

This rule reports ???.

<!-- prettier-ignore-start -->

<!-- eslint-skip -->

```md
<!-- eslint markdown-preferences/max-len: 'error' -->

<!-- âœ“ GOOD -->


<!-- âœ— BAD -->

```

<!-- prettier-ignore-end -->

## ðŸ”§ Options

This rule has no options.

<!-- or -->

```json
{
  "markdown-preferences/max-len": [
    "error",
    {}
  ]
}
```

- 

## ðŸ“š Further Reading

- 

## ðŸ‘« Related Rules

- [xxx]

[xxx]: https://xxx

