---
pageClass: "rule-details"
sidebarDepth: 0
title: "markdown-preferences/padded-custom-containers"
description: "disallow or require padding inside custom containers"
---

# markdown-preferences/padded-custom-containers

> disallow or require padding inside custom containers

- ❗ <badge text="This rule has not been released yet." vertical="middle" type="error"> **_This rule has not been released yet._** </badge>
- ⚙️ This rule is included in `plugin.configs.standard`.
- 🔧 The `--fix` option on the [command line](https://eslint.org/docs/user-guide/command-line-interface#fixing-problems) can automatically fix some of the problems reported by this rule.

## 📖 Rule Details

This rule reports ???.

<!-- prettier-ignore-start -->

<!-- eslint-skip -->

```md
<!-- eslint markdown-preferences/padded-custom-containers: 'error' -->

<!-- ✓ GOOD -->


<!-- ✗ BAD -->

```

<!-- prettier-ignore-end -->

## 🔧 Options

This rule has no options.

<!-- or -->

```json
{
  "markdown-preferences/padded-custom-containers": ["error", {}]
}
```

-

## 📚 Further Reading

-

## 👫 Related Rules

- [xxx]

[xxx]: https://xxx

## 🔍 Implementation

<!-- eslint-disable markdown-links/no-dead-urls -- Auto generated -->

- [Rule source](https://github.com/ota-meshi/eslint-plugin-markdown-preferences/blob/main/src/rules/padded-custom-containers.ts)
- [Test source](https://github.com/ota-meshi/eslint-plugin-markdown-preferences/blob/main/tests/src/rules/padded-custom-containers.ts)
- [Test fixture sources](https://github.com/ota-meshi/eslint-plugin-markdown-preferences/tree/main/tests/fixtures/rules/padded-custom-containers)

<!-- eslint-enable markdown-links/no-dead-urls -- Auto generated -->
